src/sympc/protocol/spdz/spdz.py

```diff
@@ -28,40 +39,65 @@ def mul_master(x, y, op_str):
     eps = x - a_sh
     delta = y - b_sh
     session = x.session
-    nr_parties = len(session.session_ptr)
+    nr_parties = len(session.session_ptrs)
 
     eps_plaintext = eps.reconstruct(decode=False)
     delta_plaintext = delta.reconstruct(decode=False)
 
-    with ThreadPoolExecutor(max_workers=nr_parties) as executor:
-        args = list(zip(session.session_ptr, a_sh.shares, b_sh.shares, c_sh.shares))
-        futures = [
-            executor.submit(
-                session.parties[i].sympc.protocol.spdz.mul_parties,
-                *args[i],
-                eps_plaintext,
-                delta_plaintext,
-                op_str,
-            )
-            for i in range(nr_parties)
-        ]
-
-    shares = [f.result() for f in futures]
+    args = list(
+        map(
+            list,
+            zip(
+                session.session_ptrs, a_sh.share_ptrs, b_sh.share_ptrs, c_sh.share_ptrs
+            ),
+        )
+    )
```

Comment:
Haha this is a bit ugly! But maybe ok for now

Reply:
Made it cleaner